src/vs/workbench/api/common/extHostTerminalService.ts

```diff
@@ -794,8 +795,11 @@ export abstract class BaseExtHostTerminalService extends Disposable implements I
 		if (completions === null || completions === undefined) {
 			return;
 		}
-
-		return completions;
+		if (Array.isArray(completions)) {
+			return completions;
+		} else {
+			return new TerminalCompletionList(completions.items, { ...completions.resourceRequestConfig, pathSeparator: isWindows ? '\\' : '/' });
```

Comment:
I think we want to convert this to a DTO in extHost.protocol.ts so we're not passing an object over IPC which could serialize incorrectly